bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WLedHandler.java

```diff
@@ -0,0 +1,442 @@
+                bigTemp = ((State) command).as(PercentType.class).toBigDecimal().multiply(BIG_DECIMAL_2_55);
+                sendGetRequest("/win&SX=" + bigTemp);
+                break;
+            case CHANNEL_INTENSITY:
+                bigTemp = ((State) command).as(PercentType.class).toBigDecimal().multiply(BIG_DECIMAL_2_55);
+                sendGetRequest("/win&IX=" + bigTemp);
+                break;
+            case CHANNEL_SLEEP:
+                if (OnOffType.ON.equals(command)) {
+                    sendGetRequest("/win&NL=1");
+                } else {
+                    sendGetRequest("/win&NL=0");
+                }
+                break;
+            case CHANNEL_PRESETS:
+                sendGetRequest("/win&PL=" + command);
+                break;
+            case CHANNEL_PRESET_DURATION:
+                if (command instanceof QuantityType) {
+                    QuantityType<?> seconds = ((QuantityType<?>) command).toUnit(SmartHomeUnits.SECOND);
+                    if (seconds != null) {
+                        bigTemp = new BigDecimal(seconds.intValue()).multiply(new BigDecimal(1000));
+                        sendGetRequest("/win&PT=" + bigTemp.intValue());
+                    }
+                }
+                break;
+            case CHANNEL_TRANS_TIME:
+                if (command instanceof QuantityType) {
+                    QuantityType<?> seconds = ((QuantityType<?>) command).toUnit(SmartHomeUnits.SECOND);
+                    if (seconds != null) {
+                        bigTemp = new BigDecimal(seconds.intValue()).multiply(new BigDecimal(1000));
+                        sendGetRequest("/win&TT=" + bigTemp.intValue());
+                    }
+                }
+                break;
+            case CHANNEL_PRESET_CYCLE:
+                if (OnOffType.ON.equals(command)) {
+                    sendGetRequest("/win&CY=1");
+                } else {
+                    sendGetRequest("/win&CY=0");
+                }
+                break;
+        }
+    }
+
+    public void savePreset(int presetIndex) {
+        if (presetIndex > 16) {
+            logger.warn("Presets above 16 do not exist, and the action sent {}", presetIndex);
+            return;
+        }
+        sendGetRequest("/win&PS=" + presetIndex);
+    }
+
+    private void pollLED() {
+        sendGetRequest("/win");
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(WLedConfiguration.class);
+        if (!config.address.contains("://")) {
+            logger.debug("Address was not entered in correct format, it may be the raw IP so adding http:// to start");
+            config.address = "http://" + config.address;
+        }
+        pollingFuture = scheduler.scheduleWithFixedDelay(this::pollLED, 1, config.pollTime, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingFuture != null) {
+            pollingFuture.cancel(true);
+        }
```

Comment:
Please address the null build warnings. If there are any others please address them by using a local variable like so:
```suggestion
        Future<?> future = pollingFuture;
        if (future != null) {
            future.cancel(true);
        }
```

Reply:
Thank you for showing how it is meant to be handled, all warnings of that type are gone.